packages/app-mobile/components/NotesBar.tsx

```diff
@@ -0,0 +1,282 @@
+				backgroundColor: theme.dividerColor,
+				height: 1,
+				width: '100%',
+			},
+			nativeInput: {
+				fontSize: theme.fontSize,
+				flex: 1,
+				paddingRight: 8,
+			},
+			searchIcon: {
+				fontSize: 22,
+			},
+			searchInput: {
+				alignItems: 'center',
+				backgroundColor: theme.backgroundColor,
+				paddingLeft: 8,
+				borderRadius: 4,
+				borderWidth: 1,
+				borderColor: theme.dividerColor,
+				height: 42,
+				flex: 1,
+			},
+			button: {
+				height: 42,
+				width: 42,
+				backgroundColor: theme.color4,
+				alignItems: 'center',
+				justifyContent: 'center',
+				borderRadius: 4,
+				flex: 0.5,
+				marginLeft: 8,
+			},
+			buttonIcon: {
+				color: theme.backgroundColor,
+				fontSize: 22,
+			},
+			inputGroup: {
+				justifyContent: 'space-between',
+			},
+		};
+	};
+
+	const styles = useStyles(stylingFunction, themeId);
+
+	const titleComp = (
+		<View style={[styles.title, styles.horizontalFlex]}>
+			<Icon name='md-document'style={[styles.top, styles.titleIcon]} />
+			<Text style={[styles.top, styles.titleText]}>{_('Notes')}</Text>
+		</View>
+	);
+
+	const dividerComp = (
+		<View style={styles.divider}></View>
+	);
+
+	const closeButtonComp = (
+		<TouchableOpacity onPress={props.toggleNotesBar}>
+			<Icon name="close" style={[styles.top, styles.closeIcon]}/>
+		</TouchableOpacity>
+	);
+
+	const renderIconButton = (icon: JSX.Element, onPress: ()=> Promise<void>) => {
+		return (
+			<TouchableOpacity style={styles.button} activeOpacity={0.8} onPress={onPress}>{icon}</TouchableOpacity>
+		);
+	};
+
+
+	const handleNewNote = async (isTodo: boolean) => {
+		let folderId = props.selectedFolderId !== Folder.conflictFolderId() ? props.selectedFolderId : null;
+		if (!folderId) folderId = props.activeFolderId;
+
+		props.dispatch({
+			type: 'NAV_BACK',
+		});
+
+		const newNote = await Note.save({
+			parent_id: folderId,
+			is_todo: isTodo ? 1 : 0,
+		}, { provisional: true });
+
+		props.dispatch({
+			type: 'NAV_GO',
+			routeName: 'Note',
+			noteId: newNote.id,
+		});
+	};
+
+	const addNoteButtonComp = renderIconButton(<Icon name='document-text-outline' style={styles.buttonIcon} />, () => handleNewNote(false));
+	const addTodoButtonComp = renderIconButton(<Icon name='checkbox-outline' style={styles.buttonIcon} />, () => handleNewNote(true));
+
+	const topComp = (
+		<View>
+			<View style={[styles.topContainer, styles.horizontalFlex]}>
+				{titleComp}
+				{closeButtonComp}
+			</View>
+			{dividerComp}
+		</View>
+	);
+
+	const handleQuerySubmit = async () => {
+		if (!query) {
+			setNotes(props.items);
+			return;
+		}
+
+		let searchQuery = query.trim();
+
+		if (searchQuery === '') {
+			setNotes(props.items);
+			return;
+		}
+
+		searchQuery = searchQuery.toLowerCase();
+
+		// Find notes in folder using only note title
+		const result = props.items.filter(item => {
+			let noteTitle = Note.displayTitle(item);
+			noteTitle = noteTitle.toLowerCase();
+
+			return noteTitle.includes(searchQuery);
```

Comment:
Search should be done using the search engine. It would be too hard to have to duplicate search logic across the codebase. Like now you're not handling accents, etc.